hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -289,17 +290,35 @@ private <BuilderT extends S3BaseClientBuilder<BuilderT, ClientT>, ClientT> void
     builder.fipsEnabled(fipsEnabled);
 
     if (endpoint != null) {
+      boolean overrideEndpoint = true;
       checkArgument(!fipsEnabled,
           "%s : %s", ERROR_ENDPOINT_WITH_FIPS, endpoint);
-      builder.endpointOverride(endpoint);
       // No region was configured, try to determine it from the endpoint.
       if (region == null) {
-        region = getS3RegionFromEndpoint(parameters.getEndpoint());
+        boolean endpointEndsWithCentral =
```

Comment:
so from the response on SDK issue, it looks like we don't want to override if endpoint is `s3.amazonaws.com` and region is not `US_EAST_1`. 

Currently, if `fs.s3a.endpoint` is `s3.amazonaws.com` and `fs.s3a.endpoint.region` is `eu-west-1` for example, we will end up overriding and end up with the same 400 errors. 

What if we never override if endpoint is s3.amazonaws.com?

Reply:
> What if we never override if endpoint is s3.amazonaws.com?

That sounds right, let me test with various combinations of endpoint and region before making the changes:

e.g.
1. endpoint central and region null
2. endpoint central and region anything other than us-east-2
3. endpoint central and region us-east-2
4. endpoint null and region null
5. endpoint s3-us-east-2.amazonaws.com and region us-east-2 (and null)
6. endpoint s3-us-east-1.amazonaws.com and region us-east-1 (and null)